src/gluonts/model/estimator.py

```diff
@@ -182,13 +188,15 @@ def train_model(
             train_iter=training_data_loader,
         )
 
-        return transformation, trained_net
+        with self.trainer.ctx:
+            # ensure that the prediction network is created within the same MXNet
+            # context as the one that was used during training
+            return TrainOutput(
+                transformation=transformation,
+                trained_net=trained_net,
+                predictor=self.create_predictor(transformation, trained_net),
```

Comment:
Why including the `predictor` here? It is a function of the other two fields (`transformation` and `trained_net`) so maybe having it as a separate field introduces redundancy